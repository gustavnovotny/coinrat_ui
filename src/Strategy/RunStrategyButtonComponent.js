// @flow
import React from "react"
import {Button, Position, Tooltip} from "@blueprintjs/core"

type Props = {
  onClick: () => void,
}

const RunStrategyButtonComponent = ({onClick}: Props) => {
  return <Tooltip content="Run simulation of strategy on given time range" position={Position.BOTTOM}>
    <Button
      style={{marginTop: 1 + 'px', marginLeft: 7 + 'px'}}
      className="pt-intent-primary"
      iconName="pt-icon-play"
      onClick={onClick}
    />
  </Tooltip>
}

export default RunStrategyButtonComponent
