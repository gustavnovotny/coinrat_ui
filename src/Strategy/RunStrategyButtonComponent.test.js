// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import RunStrategyButtonComponent from "./RunStrategyButtonComponent"

it('renders correctly', () => {
  const tree = renderer.create(<RunStrategyButtonComponent onClick={() => undefined}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
