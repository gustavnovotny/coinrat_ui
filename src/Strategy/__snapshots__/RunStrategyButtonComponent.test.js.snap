// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span>
  <div
    className="pt-popover-target"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <button
      className="pt-button pt-icon-play pt-intent-primary"
      disabled={undefined}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "marginLeft": "7px",
          "marginTop": "1px",
        }
      }
      tabIndex={0}
      type="button"
    />
  </div>
</span>
`;
